File: go/internal/scoring/testdata/structural-v1/go/state_cosmetic_after.go
package calibration

type CosmeticAfter struct{ total, updates int }

func acceptCosmetic(delta int) bool {
	if delta < 0 {
		return false
	}
	return true
}
func setCosmetic(c *CosmeticAfter, delta int) {
	c.total += delta
	if c.total > 100 {
		c.total = 100
	}
}
func markCosmetic(c *CosmeticAfter) { c.updates++ }

func (c *CosmeticAfter) Advance(delta int) int {
	if !acceptCosmetic(delta) {
		return c.total
	}
	setCosmetic(c, delta)
	markCosmetic(c)
	return c.total
}

File: go/internal/scoring/testdata/structural-v1/java/StateCosmeticAfter.java
final class StateCosmeticAfter {
  private int total;
  private int updates;

  private boolean acceptState(int delta) {
    if (delta < 0) return false;
    return true;
  }
  private void setState(int delta) {
    total += delta;
    if (total > 100) total = 100;
  }
  private void markState() { updates++; }

  int advance(int delta) {
    if (!acceptState(delta)) return total;
    setState(delta);
    markState();
    return total;
  }
}
